Library/Homebrew/requirements/gpg_requirement.rb

```diff
@@ -0,0 +1,29 @@
+require "requirement"
+
+class GPGRequirement < Requirement
+  fatal true
+  default_formula "gnupg2"
+
+  satisfy(:build_env => false) { gpg2 || gpg1 }
+
+  # Homebrew installs GnuPG 1.x as `gpg` but this may change in future.
+  # MacGPG2/GPGTools installs GnuPG 2.0.x as a vanilla `gpg` symlink
+  # pointing to `gpg2`. Ensure we're not using unexpected gpg version.
+  def gpg1
+    which_all("gpg").detect do |gpg|
+      gpg_version = Utils.popen_read(gpg, "--version")[/\d\.\d/, 0]
+      next unless gpg_version
+      Version.new(gpg_version.to_s) == Version.new("1.4")
+    end
+  end
+
+  def gpg2
+    which_all("gpg2").detect do |gpg2|
+      gpg2_version = Utils.popen_read(gpg2, "--version")[/\d\.\d/, 0]
+      next unless gpg2_version
+      # For now, to give upstreams time to adjust, only support 2.0.x
+      # rather than the 2.1.x "modern" series.
+      Version.new(gpg2_version.to_s) == Version.new("2.0")
+    end
+  end
```

Comment:
> Note that we're only comparing the first two numbers,

It's not very abvious here.Maybe call the variable `gpg_short_version`.

>  and there will never be a `gpg` 1.5 release, 

I think a comment is required in the code.

> and we're intentionally avoiding `2.1` here.

Same as above, a comment is needed for better understanding. However, I don't think current `==2.0` is a right approach for following reasons:

* It would be weird and unexpected to users. For users who have `gpg 2.1` installed, it almost seems Homebrew cannot find gpg. 
* It fails to consider future 2.2+ release?
* What about when upstream fixes the issue?

The latter two reasons can be very harmful if a user in the future somehow stucks in the old version of brew code. In that case, we are forcing them to use old and obsolete gpg.

May I suggest something like?
```ruby
gpg2_version >= "2.0" && !known_break_releases.include?(gpg2_version)